src/tracer.js

```diff
@@ -246,7 +246,9 @@ export default class Tracer {
       hasValidParent = true;
       let spanId = Utils.getRandom64();
       ctx = parent._makeChildContext(spanId);
-      ctx.finalizeSampling(); // will finalize sampling for all spans sharing this traceId
+      if (parent.isRemote()) {
+        ctx.finalizeSampling(); // will finalize sampling for all spans sharing this traceId
```

Comment:
IMHO the comment is redundant when the reader knows ctx is shared

Reply:
and if the reader forgets it, here's a reminder which is critical to understanding this piece